### ISSUE TITLE
ID 259326: [Screen reader -Simon - Rate then Sum]: Heading hierarchy is not defined sequentially in the "Explore" Screen.

### ISSUE THREAD

**Issue by natydej** [grafana/grafana#66560]
### **User Experience:** 
Visual Impaired users will be unable to understand and face difficulty in relating the content while navigating the page in the heading mode of the screen reader as the heading structure/order is not logical.
_Note: User credentials should NOT be included in the bug._

### **Repro steps:**

1. Open [https://play.grafana.org/](https://play.grafana.org/) Sign in with valid credentials. "Home" page appears. 
2. Navigate to the left navigation bar and to the "Explore" button, select it. 
3. New **"Explore"** page appears, Navigate it.  
4. Navigate to the **"Query pane's"** first row and expand it.  
5. Navigate to the **"Query Patterns"** combo box and select **"Rate the sum".** 
6. New block gets added in operations. Navigate the **"Sum"** block. Verify the issue

**Actual Result:** 
 Issue: Heading hierarchy is not defined sequentially in the "Explore" Screen.

**Observation**: 
 Heading hierarchy is not sequential in the "Explore" screen as 'Options' is defined under `<H6>` Heading level just after `<H1>` heading text "Explore". Due to which Screen reader announces as "Heading level 6 options ".

**Expected Result:** 

The heading hierarchy should be defined sequentially in the **"Explore"** Screen as the 'Options " should be defined under `<H2>` heading level which is just after `<H1>` heading text “Explore”.

_Note:  Same issue throughout the application._

### **Severity:**
`SEV2`

### **Area:**
`Query Editor`

**Comment by ifrost** [comment]
Which data source is causing the problem? Query Editors are part of data source plugins. This issue may need to be re-assigned to the plugin maintainer.

**Comment by Elfo404** [comment]
@grafana/observability-logs maybe this is related to Loki's query patterns? assigning this to you, feel free to bounce it back

**Comment by gabor** [comment]
hmm. i cannot really reproduce it, closest i got is:
1. open `explore`
2. choose a prometheus datasource
3. choose `kick start your query`
4. choose `rate query starters`
5. choose `rate then sum`

so assigning this to @grafana/observability-metrics .

but, things have changed and "query patterns" do not exist anymore, it's "kickstart your query" now with a different UI.

@natydej sorry, could you please check, does the problem still exist on the current https://play.grafana.org site?

**Comment by natydej** [comment]
Here is a video that shows better what the current issue is, as you can see that's an example from Prometheus data source

https://user-images.githubusercontent.com/51436795/233681887-b4ec46fc-c360-4476-ba6c-36188dcd0172.mov

**Comment by gabor** [comment]
@natydej thank you for the additional info. as this video deals with the Prometheus data source, i will assign this for now to @grafana/observability-metrics (feel free to reassign it if it is incorrect). thanks!

**Comment by gtk-grafana** [comment]
Looks like we've got a few h6 for this title in a few datasources, not just prom:

* https://github.com/grafana/grafana/tree/main/grafana/grafana/public/app/plugins/datasource/parca/QueryEditor/QueryOptions.tsx#L45
* https://github.com/grafana/grafana/tree/main/grafana/grafana/public/app/plugins/datasource/phlare/QueryEditor/QueryOptions.tsx#L62
* https://github.com/grafana/grafana/tree/main/grafana/grafana/public/app/plugins/datasource/prometheus/querybuilder/shared/QueryOptionGroup.tsx#L33

@natydej is the solution here to just remove the h6 and use a div instead?

**Comment by gtk-grafana** [comment]
Also @gabor it looks as though that prometheus component is shared with Loki, so this is a problem with Loki as well (but metrics can fix this one).

@natydej I removed the headers locally and now I'm just not seeing the headers option in the screen reader at all, not sure if that's the desired functionality or not.

Let me know if you have a few minutes to jump on a call and get me pointed in the right direction.

**Comment by GunFood** [comment]
Any Update on this issue?

**Comment by matyax** [comment]
@GunFood are you experiencing any issue with screen readers and query editors? If you can share some details about it, it would be very helpful for the resolution.

**Comment by GunFood** [comment]
Hi @matyax. 
Not really. I'm currently in a software introduction process in the company and all parties have to be heard. Our accessibility
manager (don't really know the english word) send a list with open Github issues regarding these topics and would like to see them solved for his final Go for Grafana in the company or at least a plan than they might be solved. We plan to buy Grafana Enterprise and in public sector it is quiet difficult with a lot of regulations.
That's the reason I've asked on several tickets today for a status update.
